Tools/build/smelly.py

```diff
@@ -7,7 +7,7 @@
 import sysconfig
 
 
-ALLOWED_PREFIXES = ('Py', '_Py')
+ALLOWED_PREFIXES = ('Py', '_Py', 'mi_', '_mi_')
```

Comment:
It's kind of unfortunable :-( These symbols should not exported, no?

Reply:
Looks like Christian solved this by using mimalloc's `static.c` and a set of re-defs of the exports to be `_Py` prefixed,  I'll re-use that approach (and prefix with `PyUnstable`)

Reply:
Would it be possible to have a script which imports mimalloc and applies a patch on top of it to rename mi and _mi symbols to add a _Py prefix? Or at least a patch?

We do have a patch on on top of Python/dtoa.c which is maintained Python for example. My notes on vendored libraries: https://pythondev.readthedocs.io/files.html#vendored-external-libraries

Reply:
@vstinner Christian included the command he used in his `pycore_mimalloc.h` and it just produces a bunch of `#define`s to do the renames, and then no patches are necessary to mimalloc.  It seems like a pretty clean solution to me?

@colesbury Ahh, you're right...  I'll just re-drop `static.c` and do the same set of .c `#include`s it does in `obmalloc.c` minus the alloc override ones.  We'll just need to update things with upgrades if the file set changes.

Reply:
Do we have to **export** these mi_alloc() and mi_free() symbols? Can't we make them private, and let C extensions call PyMem_Malloc() and PyMem_Free()?

Reply:
So I don't think anything is actually exported here with the current build.  If you do `./configure --enable-shared` and run `nm libpython3.13.so.1.0` all of the mi_malloc related symbols are of type `t` meaning they are not global.

Likewise if you run `nm ./python` you'll see that in the statically linked `python` that all of the symbols are private as well.

It's only when we run `nm` on the static library directly that they're still listed as being global.  I assume this is some artifact of the fact that this still needs to be linked in and things will get updated after that - but I must admit I don't fully understand this.

I think this is mainly a consequence of Python's build specifying that default visibility is hidden.  mimalloc will specifically mark symbols it wants as exported with `__attribute__((visibility("default")))` when it wants them exported (and on Windows will do `__declspec(dllexport)`).  

The big reason for the additional symbols that smelly is picking up though is that CPython uses `static` very liberally throughout the code base and mimalloc spreads its implementation across multiple files not using `static`. 

I think probably the most straight forward fix here is to allow `_mi` and `mi` prefixes for only static library, and disallow them for the shared library.  This will assure that there are no exported symbols that shouldn't be there but will also keep the good hygine of using `static` where appropriate in the CPython code base.